### PR TITLE
[script][common-crafting] Cost calculator for crafting

### PR DESCRIPTION
**Pull request by Raykyn55**
Overview:
Adds crafting_cost method to common-crafting, which utilizes changes to base-recipes and base-crafting to draw parts lists, parts costs, and materials costs, to calculate an exact cost for a given craft routine.

Why:
With the changes to workorders materials, some of the choices are more expensive, and require more cash on hand to complete your workorder. Furthermore, each craft script that handles grunt scripts (craft,smith,eventually others) has numerous calls to common-money for amounts which are rough estimates of what's required. These estimates ignore denomination and, since the workorders change, often too low even then. This would allow for handler scripts to simply send the relevant information to this method, and get back an exact cost for determining whether we have enough coin on hand.

Major changes:
Added to 'stock' section of base-crafting to include enchanting items, various parts, and the costs associated with all of them, in Kronars.
Rearranged base-recipes under discipline headings (blacksmithing,remedies,artificing,etc)
Within these headings, each recipe loses 'type', as this is functionally now the header.
Each recipe has it's parts organized under 'part' for easy reference.

Attempted to capture all the relevant references, sure I probably missed one(or three).

### CODE REVIEW

**Comment by MahtraDR** [comment]
Can we break this up into data changes and then script changes? Adding data is harmless. We can then test the scrip changes as we go. Or even one at a time.

**Comment by Raykyn55** [comment]
> Can we break this up into data changes and then script changes? Adding data is harmless. We can then test the scrip changes as we go. Or even one at a time.

We can add data to base-crafting in it's own PR, but the changes to base-recipes can't be done separately.

File: workorders.lic [hunk 1]
```diff
@@ -686,21 +684,13 @@ class WorkOrders
...
-    # Sigil 3
-    DRCC.order_enchant(info['stock-room'], quantity, recipe['enchant_stock3'], @bag, @belt) unless recipe['enchant_stock3'].nil?
-
-    # Enchant component #1
-    DRCC.order_enchant(info['stock-room'], quantity, recipe['item'], @bag, @belt) unless recipe['item'].nil?
-
-    # Parts #
-    order_parts(recipe['part'], quantity) if recipe['part']
+    recipe['part'].each { |part| 
+                        if ["sigil-scroll", "bone totem"].include?(part)
```
Comment (Raykyn55) [inline workorders.lic:688]: not sure if that's going to match correctly.

**Comment by Dartellum** [comment]
We have `workorder_cash_on_hand: 10000` as we left it on the user to have the proper money on hand. I understand the what this change does as it would use the exact money getting routine (I hope it would since we have one). The 10000 is what I have and is not in base. Base has `workorder_cash_on_hand:`.

**Comment by Raykyn55** [comment]
> We have `workorder_cash_on_hand: 10000` as we left it on the user to have the proper money on hand. I understand the what this change does as it would use the exact money getting routine (I hope it would since we have one). The 10000 is what I have and is not in base. Base has `workorder_cash_on_hand:`.

The problem is bigger orders with silk, cougar, or store steel can quickly exceed 1p, plus consumables which dont have a cash check. Run a workorder for 3 sets of leathers in cougar to see what i mean.

**Comment by Dartellum** [comment]
> > We have `workorder_cash_on_hand: 10000` as we left it on the user to have the proper money on hand. I understand the what this change does as it would use the exact money getting routine (I hope it would since we have one). The 10000 is what I have and is not in base. Base has `workorder_cash_on_hand:`.
> 
> The problem is bigger orders with silk, cougar, or store steel can quickly exceed 1p, plus consumables which dont have a cash check. Run a workorder for 3 sets of leathers in cougar to see what i mean.

I am not sure if you were around, or if this still occurs, for the people stealing your money in the crafting Societies. We had to grab more money than necessary to have money to purchase the items.

**Comment by Raykyn55** [comment]
> > > We have `workorder_cash_on_hand: 10000` as we left it on the user to have the proper money on hand. I understand the what this change does as it would use the exact money getting routine (I hope it would since we have one). The 10000 is what I have and is not in base. Base has `workorder_cash_on_hand:`.
> > 
> > 
> > The problem is bigger orders with silk, cougar, or store steel can quickly exceed 1p, plus consumables which dont have a cash check. Run a workorder for 3 sets of leathers in cougar to see what i mean.
> 
> I am not sure if you were around, or if this still occurs, for the people stealing your money in the crafting Societies. We had to grab more money than necessary to have money to purchase the items.

Not surprised, but this calls immediately before purchase, so the grief window is pretty small.

**Comment by Raykyn55** [comment]
Going to split this up a little.

**Pull request closed without merging**